src/turnkeyml/cli/report.py

```diff
@@ -78,11 +78,34 @@ def summary_spreadsheets(args) -> None:
 
                         # Copy the build-specific stats
                         for key, value in build.items():
-                            # Break each value in "completed build stages" into its own column
-                            # to make analysis easier
+                            # Break each value in "completed build stages" into status and duration
+                            # to make the analysis of this data easier
                             if key == fs.Keys.COMPLETED_BUILD_STAGES:
-                                for subkey, subvalue in value.items():
-                                    evaluation_stats[subkey] = subvalue
+                                previous_state_incomplete = False
+                                for stage in build[fs.Keys.SELECTED_SEQUENCE_OF_STAGES]:
+                                    duration_column_name = f"stage_duration: {stage}"
+                                    state_column_name = f"stage_status: {stage}"
+                                    if stage in build[fs.Keys.COMPLETED_BUILD_STAGES]:
+                                        evaluation_stats[
+                                            state_column_name
+                                        ] = "COMPLETED"
+                                        evaluation_stats[duration_column_name] = build[
+                                            fs.Keys.COMPLETED_BUILD_STAGES
+                                        ][stage]
+                                    elif not previous_state_incomplete:
+                                        previous_state_incomplete = True
+                                        evaluation_stats[
+                                            state_column_name
+                                        ] = "INCOMPLETE"
+                                        evaluation_stats[duration_column_name] = "-"
+                                    else:
+                                        evaluation_stats[
+                                            state_column_name
+                                        ] = "NOT STARTED"
+                                        evaluation_stats[duration_column_name] = "-"
+
+                                # Do not add the raw version of COMPLETED_BUILD_STAGES to the report
+                                continue
```

Comment:
I'm not sure this is the best implementation of the desired logic. Done this way, only the report.csv will have the `stage_status` and `stage_duration` members, and the `turnkey_stats.yaml` will have something that looks pretty different.

It should be pretty easy to only record the stats we want, and to have this be the same across both the stats file and the report file, without the need for extra decoding logic here.

It would work like this. [In the code block](https://github.com/onnx/turnkeyml/blob/a52414c1abb46782f2b37c9d8b60cd8b145496e0/src/turnkeyml/build/stage.py#L289C22-L289C22) where we are currently setting `COMPLETED_BUILD_STAGES`, do the following instead:
1. Before running any stages, loop over the sequence and `save_model_eval_stat(STAGE_NAME, NOT_STARTED)`
2. Just before a stage starts, call `save_model_eval_stat(STAGE_NAME, INCOMPLETE)`
3. When the stage ends, call `save_model_eval_stat(STAGE_NAME, COMPLETED)`

Can do something similar for `stage_duration` as well. Although I would suggest actually setting the `stage_duration` in a `finally` block so that we get the duration on failure cases as well (which is often more interesting).

Reply:
Great suggestion. I changed things as you recommended.